### PR TITLE
Use FullName as service name. Fixes #373

### LINKED ISSUE #373: Cannot register two different types with same name when assembly scanning with predicate
I am using LightInject to register mapping profiles for AutoMapper. I have code like the following

```csharp
public class Foo {
  ...
  public class MappingProfile : Profile {
    ...
  }
}

public class Bar {
  ...
  public class MappingProfile : Profile {
     ...
  }
}

public class TestBase {
  public TestBase() {
    var container = new ServiceContainer();
    container.RegisterAssembly(typeof(Foo).Assembly, (serviceType, _) => serviceType == typeof(Profile));
    var profiles = container.GetAllInstances<Profile>();
  }
}
```

When I run the code and examine `profiles`, only `Foo.MappingProfile` is present.

If I rename `Bar.MappingProfile` to `BarMappingProfile`, both services are present.

### PR DESCRIPTION
**Pull request by geirsagberg**
Fixes #373

### CODE REVIEW

File: src/LightInject/LightInject.cs [hunk 1]
```diff
@@ -6212,16 +6212,16 @@ public void Scan(Assembly assembly, IServiceRegistry serviceRegistry)
...
             if (implementingType.GetTypeInfo().IsGenericTypeDefinition)
             {
-                var regex = new Regex("((?:[a-z][a-z]+))", RegexOptions.IgnoreCase);
+                var regex = new Regex("((?:[a-z][a-z.]+))", RegexOptions.IgnoreCase);
                 implementingTypeName = regex.Match(implementingTypeName).Groups[1].Value;
                 serviceTypeName = regex.Match(serviceTypeName).Groups[1].Value;
             }
 
-            if (serviceTypeName.Substring(1) == implementingTypeName)
+            if (serviceTypeName.Split('.').Last().Substring(1) == implementingTypeName.Split('.').Last())
```
Comment (geirsagberg) [inline src/LightInject/LightInject.cs]: Not sure if this has any performance considerations?

Comment (seesharper) [inline src/LightInject/LightInject.cs]: We'll leave it like this for now. Should not have a deep impact on performance

**Comment by seesharper** [comment]
Thanks a lot again for you time and effort

**Comment by geirsagberg** [comment]
No problem, happy to help :)

**Pull request merged**